### DISCUSSION TITLE
ROG Ally support?

### DISCUSSION THREAD

**Discussion by aLanaMaunz** [seerge/g-helper#999] [General]
**Is your feature request related to a problem? Please describe.**
Better support for the ROG Ally, right now G-Helper works pretty well but some features are not working.

**Describe the solution you'd like**
Removal of the d-GPU section and fixes for the button mapping. 

**Describe alternatives you've considered**
Using another tool only for button mapping and TDP changes would work too, but the whole power-plan and autoswitching while docking make G-Helper super useful. 

**Additional context**
Hey Serge! Thanks a lot for this great tool :)
I don't know if this is known or even intended, but because the ROG Ally is using armory-crate and all related services too (and has the same endpoints?) G-Helper already works pretty good. But I think it could need some minor improvements or fixes. 
The d-GPU section (because the Ally only has a iGPU) is not needed at all, at least if you don't use a XG-mobile? Don't know if XG-mobile even works with G-Helper. Maybe an option to hide the whole section and resize the GUI would be enough.
The hotkey settings are also somehow half-working. M1 and M2 are the "volume +" and "volume -" buttons (remapping is possible and fun, but volume control is already a good usage for these buttons.), and besides this only the ROG-button can be remapped. Disabling the ASUS services disables the "command-center" (part of AC suite) overplay button, but it seems that this one cant be remapped with G-Helper. A way to remap the ROG-button while keeping the ASUS services running, which opens AC by default, to something else would fix a lot of issues because the ASUS services are still somehow needed on the Ally for useful things like gamepad-desktop-switching and stuff like that.

I know G-Helper has been created for the ROG laptops, but with increasing sales of the Ally it would be great to see better support. G-Helper is already a great tool and fixes the "TDP issues" on my Ally pretty well - thanks a lot!

**Comment by seerge** [comment]
@aLanaMaunz hello. I'm aware that G-Helper works on Ally :) 

>But I think it could need some minor improvements or fixes.
The d-GPU section (because the Ally only has a iGPU) is not needed at all, at least if you don't use a XG-mobile? Don't know if XG-mobile even works with G-Helper. Maybe an option to hide the whole section and resize the GUI would be enough.

- XGM is supported and can be enabled / disabled via G-Helper
- GPU mode indeed is not "needed" as is, but app shows it cause Ally's BIOS reports this endpoint (for devices that don't support disabling GPU endpoint doesn't exist). Moreover - setting Standard / Eco switch even seem to do something on Ally, i just didn't figure out what ... (mb it affects VRAM dedicated to iGPU?)

> The hotkey settings are also somehow half-working. M1 and M2 are the "volume +" and "volume -" buttons (remapping is possible and fun, but volume control is already a good usage for these buttons.), and besides this only the ROG-button can be remapped. 

- M1-M4 keys under extra are for **LAPTOP M-keys**(!). And on laptops M1,M2 are simple volume keys. Those are NOT backpedals on Ally. Furthermore - both backpedals on ally emit single keycode (so they are technically one button) 

>  Disabling the ASUS services disables the "command-center" (part of AC suite) overplay button, but it seems that this one cant be remapped with G-Helper. A way to remap the ROG-button while keeping the ASUS services running, which opens AC by default, to something else would fix a lot of issues because the ASUS services are still somehow needed on the Ally for useful things like gamepad-desktop-switching and stuff like that.

G-Helper has never been intended to run all the time along AC. There is no way to remap ROG button w/o stopping asus service (as it literally listens to that button and runs AC on click)

I may add some basic bindings for backpedals on Ally later, but please keep in mind that G-Helper is currently being used by ~ 20-25K people (by my  estimations), and Ally users are only very small part of them https://github.com/seerge/g-helper/discussions/707 (probably like 1-2% max)

AC on Ally does a lot of "extra" things that are not needed on laptops such controller support, AMD driver settings (FSR / FPS limit / etc), and many more. Adding all this functionality just for 1-2% of users is quite a lot for a **single** developer as me (who does all this voluntarily and in a free time).

**Comment by davnozdu** [comment]
Possible to add AutoTDP function for Rog Ally ?

**Comment by seerge** [comment]
@aLanaMaunz you may check latest 0.105 as it has ability to bind command center and backpedal buttons https://github.com/seerge/g-helper/releases

And I'm moving this to discussion, as this is not really a feature request anyway.

**Reply by aLanaMaunz** [reply]
Thanks a lot~! ^_^
